Add no header

MartinFillon opened the pull request:
closes #766 

also relates to #761  as I created the branch from it

PThorpe92 commented:
commit msg should probably be `feat(icons)` instead of `feat(docker)`.

looking at this now

PThorpe92 commented on src/options/view.rs:
```diff
@@ -169,13 +181,28 @@ impl details::Options {
+            return Ok(false);
+        };
+
+        match word.to_string_lossy().as_ref() {
```
this is allocating unnecessarily, its calling `to_os_string()` and then again with `to_string_lossy()`, can we try something like this?
```rust
use crate::options::flags::WHEN;
```
```rust
    fn deduce_header(matches: &MatchedFlags<'_>, stdout_istty: bool) -> Result<bool, OptionsError> {
        match matches.get(&flags::HEADER)? {
            Some(word) if word == WHEN[0] => Ok(true), // always
            Some(word) if word == WHEN[1] => Ok(stdout_istty), // auto
            Some(word) if word == WHEN[2] => Ok(false), // never
            Some(other) => Err(OptionsError::BadArgument(&flags::HEADER, other.into())), // bad arg
            None => Ok(false), // no flag
        }
    }
```

PThorpe92 reviewed (changes requested):
![image](https://github.com/eza-community/eza/assets/121899304/5059a02d-8b1b-4329-8a32-3b4561880080)

`src/options/views.rs` has all the tests that need to have the new arg provided now (why the tests/CI is failing). Other than that and the `deduce` everything looks good. Looking forward to clap where we can offload all this convoluted arg parsing :sweat_smile:

MartinFillon commented:
> src/options/views.rs has all the tests that need to have the new arg provided now (why the tests/CI is failing). Other than that and the deduce everything looks good. Looking forward to clap where we can offload all this convoluted arg parsing

This is not why I am struglling with a bug gonna push the first fixes

MartinFillon commented:
poke @PThorpe92

PThorpe92 commented:
> poke @PThorpe92

my fault buddy what was the issue you were having again?

MartinFillon commented:
strictness doesnt fail as expected it goes through on a unit test from view.rs iirc

and its okay i just saw i didnt have it passing so as we rapidly talked about it i just thought when he has time he will see this one

MartinFillon commented:
Hey, just a heads up to say that this pr is ready

cafkafk reviewed (changes requested):
Seems like there's some tests that fail in CI :S

MartinFillon commented:
It should be ready

MartinFillon commented:
Hey this pr is ready for review

cafkafk reviewed (commented):
Seems like the DirEntry PR created some conflicts that need to be resolved :/

MartinFillon commented:
Hey @cafkafk and @PThorpe92 I have revived a bit this PR as I have gotten more time lately, feel free to take a look whenever you feel like it

cafkafk reviewed (changes requested):
I'm going through our backlog, currently this seems to need to be rebased on main. That however is quite a task.

MartinFillon committed:
feat(options)!: add when capabilities to header option

--header now takes a WHEN argument
it is used to disable header if the output is redirected to a tty
